### PR TITLE
writecache: Fix db counter decrement

### PR DESCRIPTION
**Pull request by ZhangTao1596**
Signed-off-by: ZhangTao1596 <zhangtao@ngd.neo.org>

@alexvanin

### CODE REVIEW

**Comment by alexvanin** [comment]
Will be included in the patch for v0.27.0 release.

**Comment by alexvanin** [comment]
Rebase it after https://github.com/nspcc-dev/neofs-node/pull/1028

File: pkg/local_object_storage/writecache/storage.go [hunk 1]
```diff
@@ -109,15 +108,16 @@ func (c *cache) deleteFromDB(keys [][]byte) error {
...
-			sz += uint64(len(has))
 			storagelog.Write(c.log, storagelog.AddressField(string(keys[i])), storagelog.OpField("db DELETE"))
 		}
 		return nil
 	})
 	if err != nil {
 		return err
 	}
-	c.objCounters.DecDB()
+	for range keys {
```
Comment (cthulhu-rider) [inline pkg/local_object_storage/writecache/storage.go:118]: It is not always possible to delete all objects by keys. I propose, taking into account the lightness of calling `c.objCounters.DecDB()`, to do it right in the transaction above after successful `b.Delete(keys[i])`. Or we can use counter and iterate until its value here.

Comment (ZhangTao1596) [reply pkg/local_object_storage/writecache/storage.go:118]: If an error occurs, will the deleted object not be recovered?

Comment (cthulhu-rider) [reply pkg/local_object_storage/writecache/storage.go:118]: Oh, yes, ur right, I missed that we only change the internal storage of the BoltDB. According to `Update` docs `the entire transaction is rolled back`, so your code is correct.

**Review by cthulhu-rider — APPROVED** [review]
@ZhangTao1596 could you please squash your commits into a single one (with issue num)?

**Comment by ZhangTao1596** [comment]
> @ZhangTao1596 could you please squash your commits into a single one (with issue num)?

👌

**Review by alexvanin — APPROVED** [review]

**Pull request merged**